discovery/kubernetes/kubernetes.go

```diff
@@ -101,19 +122,70 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
-	if c.Role == "" {
-		return errors.Errorf("role missing (one of: pod, service, endpoints, node, ingress)")
-	}
 	err = c.HTTPClientConfig.Validate()
 	if err != nil {
 		return err
 	}
 	if c.APIServer.URL == nil && !reflect.DeepEqual(c.HTTPClientConfig, config_util.HTTPClientConfig{}) {
 		return errors.Errorf("to use custom HTTP client configuration please provide the 'api_server' URL explicitly")
 	}
+	switch c.Role {
+	case "pod":
+		for _, selector := range c.Selectors {
+			if selector.Role != RolePod && selector.Field != "" {
```

Comment:
shouldn't we test for `selector.Label` too?

Reply:
Actually I think that the incorrect role being present is enough to return error, no matter if field and label are empty, so I've reimplemented it so that we check only role correctness